### ISSUE TITLE
check to see if we need all Windows compiler macros in preprocessor

### ISSUE THREAD

**Issue by houtanb** [DynareTeam/dynare#1315]
This page
http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system

Seems to say all we need is `_WIN32` as it's defined on all Windows systems. This would make `__CYGWIN32__` and `__MINGW32__` redundant. Check to see this is the case

**Comment by JohannesPfeifer** [comment]
@houtanb What's needed to close this ticket?

**Comment by houtanb** [comment]
@JohannesPfeifer the patience to replace the above definitions in the code and the ability to test them on various versions of Windows.

For future reference, here is a more succinct article: http://stackoverflow.com/questions/5919996/how-to-detect-reliably-mac-os-x-ios-linux-windows-in-c-preprocessor

**Comment by houtanb** [comment]
@JohannesPfeifer do you compile the preprocessor on MinGW? Cygwin?

**Comment by JohannesPfeifer** [comment]
@houtanb A long time ago I used Cygwin, but I haven't done that in ages. Do you need it?

**Comment by houtanb** [comment]
No, was just wondering what I can get rid of. I just need to check this with @FerhatMihoubi before making changes and closing this ticket.

@JohannesPfeifer so you don't use MinGW either?

**Comment by JohannesPfeifer** [comment]
@houtanb I use it for the `use_dll` Option, but I don't compile the preprocessor.
